modules/httplib/proxy.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package httplib
+
+import (
+	"crypto/tls"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// MakeReverseProxyHandler creates HTTP handler that reverse-proxies requests to destURL
+func MakeReverseProxyHandler(destURL, path string, skipVerify bool) http.Handler {
+	url, _ := url.Parse(destURL)
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		proxy := httputil.ReverseProxy{
+			Director: func(req *http.Request) {
+				req.URL.Scheme = url.Scheme
+				req.URL.Host = url.Host
+				req.URL.Path = url.Path + strings.TrimPrefix(req.URL.Path, path)
+			},
+			Transport: &http.Transport{
+				MaxIdleConns:        100,
```

Comment:
100 idle connections seem too many.

At most, 2~5 is enough for development.

Reply:
I'm not sure the effects of  this, but would you agree on 10?

Reply:
By my backend / server proxying experiences, I am pretty sure about 2 is good enough, at most 5.

It only affects the idle HTTP keep-alive connections.

Reply:
Set it to 5.